src/clr-angular/data/datagrid/datagrid-page-size.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, Input } from '@angular/core';
+import { Page } from './providers/page';
+
+@Component({
+  selector: 'clr-dg-page-size',
+  template: `
+    <ng-content></ng-content>
+    <div class="clr-select-wrapper">
+      <select [(ngModel)]="page.size">
+        <option *ngFor="let option of pageSizeOptions" [ngValue]="option">{{option}}</option>
+      </select>
+    </div>
+  `,
+})
+export class ClrDatagridPageSize {
+  private _pageSizeOptions: number[] = [];
+
+  constructor(public page: Page) {}
+
+  ngOnInit() {
+    if (this.pageSizeOptions.length === 0) {
+      this.pageSizeOptions = [this.page.size];
```

Comment:
@mathisscott: This is a nice fallback, a select with a single option. If you're still worried, how about we log a warning inside of this if:
`console.warn('You are using the <clr-dg-page-size> component but forgot to provide a list of acceptable page sizes.');`

Reply:
I'm just wondering if there's an easier way for people to hide the dropdown if they don't want it. If they don't set the pageSizeOptions, it's pretty clear they don't want it. So why do they need to write CSS for every time they want to hide it?

Reply:
Oh, that's the misunderstanding. If they don't put the `<clr-dg-page-size>` component, everything works with the page size they defined themselves, and there is just no dropdown or page size selection at all. The component itself is opt-in.

Reply:
Oh. Got it now. I'm sorry for the confusion.